### PR TITLE
first

### PR DESCRIPTION
**Pull request by Christopher-Bradshaw**
psql can't handle indexed cols that are larger than 8kb. If we have a payload larger than this, we log what it was and have the payload tell us to check the logs. This should not be a long term solution - once you see your job has this problem you should find a workaround, e.g. upload payload to s3. However, at least with this change you are notified of this rather than have 504 errors when gearman-admin doesn't respond.

### CODE REVIEW

**Comment by Christopher-Bradshaw** [comment]
cc @mohit the cause of the mystery 504s we talked about was that gearman-admin didn't respond when it errored putting things in its db

File: routes.coffee [hunk 1]
```diff
@@ -88,6 +94,7 @@ module.exports = (app) ->
       create_job function_string, job_data, (err, newJob) ->
         if err
           console.error "Error creating job: #{err}"
+          res.status(500).send("")
```
Comment (mohit) [inline routes.coffee]: why not send an message with 500

**Comment by mohit** [comment]
wahoo @Christopher-Bradshaw good find!

File: routes.coffee [hunk 2]
```diff
@@ -17,6 +17,12 @@ goto_last_page_if_present = (req, res) ->
     # sane default just incase
     res.redirect '/'
 
+truncate_workload = (job_data) ->
+  log_data = job_data.match(/.{1,10000}/g) # logs are truncated at 11K chars
+  _.each log_data, (log, i) ->
+    console.log "job too large. piece #{i+1} of #{log_data.length}: #{log}"
+  "Job data too long! Check logs for details"
```
Comment (nathanleiby) [inline routes.coffee]: Can we make this a kayvee log, so we can search for or alert on it?

File: test/api.coffee [hunk 3]
```diff
@@ -95,13 +95,28 @@ describe 'routes', ->
       it 'should add a job', (done) ->
         options_ref = @options
         job_name = @job_name
+        console.log @options
```
Comment (jiancheung) [inline test/api.coffee]: remove?

File: test/api.coffee [hunk 4]
```diff
@@ -95,13 +95,28 @@ describe 'routes', ->
...
           assert.equal body.function, job_name
           assert.deepEqual body.workload, options_ref.json
           done()
 
+      it 'should truncate the job_data if longer than 8KB', (done) ->
+        _.each [1..1000], (i) =>
+          @options.json["key#{i}"] = "longlonglong"
+        options_ref = @options
+        job_name = @job_name
+        console.log @options
```
Comment (jiancheung) [inline test/api.coffee]: renive?

File: routes.coffee [hunk 5]
```diff
@@ -17,6 +19,15 @@ goto_last_page_if_present = (req, res) ->
     # sane default just incase
     res.redirect '/'
 
+truncate_workload = (job_data) ->
+  log_data = job_data.match(/.{1,10000}/g) # logs are truncated at 11K chars
```
Comment (jiancheung) [inline routes.coffee]: 10k?

**Comment by jiancheung** [comment]
a nice to have would be to update the rest of the logs to be kayvee

File: routes.coffee [hunk 6]
```diff
@@ -17,6 +19,15 @@ goto_last_page_if_present = (req, res) ->
...
+  _.each log_data, (log, i) ->
+    logger.infoD "job-too-large",
+      piece:i+1
+      total_pieces: log_data.length
+      data: log
+  "Job data too long! Check logs for details"
```
Comment (jiancheung) [inline routes.coffee]: is this what we really want to replace job_data with?

File: routes.coffee (hunk 5 above)
Comment (Christopher-Bradshaw) [inline routes.coffee]: Just to be safe. There are other things in this log that will increase this up to 10.2 so we have a reasonable safety net

File: routes.coffee (hunk 6 above)
Comment (Christopher-Bradshaw) [inline routes.coffee]: An option would be to truncate, but then it is not immediately clear that you can't just repost the job. I think that this is the best

File: test/api.coffee (hunk 4 above)
Comment (Christopher-Bradshaw) [inline test/api.coffee]: oops

File: test/api.coffee (hunk 3 above)
Comment (Christopher-Bradshaw) [inline test/api.coffee]: oops

File: routes.coffee (hunk 6 above)
Comment (jiancheung) [inline routes.coffee]: adding more context to this log would also be useful (such as function_string or handle)

**Comment by jiancheung** [comment]
one last nit about adding more context to the logs but other than that, LGTM 👍

**Pull request merged**